mangaki/mangaki/models.py

```diff
@@ -110,6 +115,29 @@ class Meta:
     def get_absolute_url(self):
         return reverse('work-detail', args=[self.category.slug, str(self.id)])
 
+    def retrieve_tags(self, anidb):
+        anidb_tags_list = anidb.get(self.anidb_aid).tags
+        anidb_tags = dict((tag[0], int(tag[1])) for tag in anidb_tags_list)
+
+        tag_work = TaggedWork.objects.filter(work=self)
+        current_tags = {tagwork.tag.title: tagwork.weight for tagwork in tag_work}
+
+        deleted_tags_keys = current_tags.keys() - anidb_tags.keys()
+        deleted_tags = dict((key, current_tags[key])for key in deleted_tags_keys)
```

Comment:
Et surtout, `{key: value for item in container}` est plus lisible (et plus court) que `dict((key, value) for item in container)`.

(BTW, pas d'espace après les `{` et `[` en python)